src/tennisSimulator/Serve.java

```diff
@@ -0,0 +1,46 @@
+package tennisSimulator;
+
+public class Serve {
+
+    private final Player firstPlayer;
+    private final Player secondPlayer;
+    private final Double SERVING_FACTOR = 0.2;
+    private final Integer RATING_DIFFERENCE;
```

Comment:
You named as if it is constant but it is not, this variable differs depending on players playing the match.

What is more, this field is used only in single method, so I think you can have it initialized there.

Reply:
I changed name, but still think that it would be better to keep this as class field. Otherwise it will be initialized every time serve is simulated, but we know that rating won`t change until Match is not finished.